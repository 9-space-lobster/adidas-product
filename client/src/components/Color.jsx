import React, { Component } from 'react';
import styled from 'styled-components';

const RenderDiv = styled.div`
  margin-right: 5px;
  margin-bottom: 5px;
`;

const StyledImage = styled.div`
  border-radius: 50%;
  width: 70px;
  height: 70px;
  border: 1px solid red;
  background: url(${props => props.url});
  background-size: 100%;
`;

const StyledSelectedTick = styled.h1`

`;

class Color extends Component {
  constructor(props){
    super(props);

    this.state = {
      product: this.props.product,
      color: this.props.color,
      isSelected: this.props.isSelected,
      url: this.props.url
    }
  }
  // Update state and render if props change
  componentDidUpdate(prevProps) {
    if (this.props.product !== prevProps.product || this.props.color !== prevProps.color || this.props.isSelected !== prevProps.isSelected) {
      this.setState({
        product: this.props.product,
        color: this.props.color,
        isSelected: this.props.isSelected,
        url: this.props.url,
      })
    }

    //this.fetchImageLocation(this.props.product.colors[this.props.color].images[0]);
  }


  render() {
    return (
      <RenderDiv>
        <StyledImage url={this.state.url}/>
      </RenderDiv>
    );
  }
}

export default Color;