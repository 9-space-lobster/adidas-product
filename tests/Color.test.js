import React from 'react';
import { shallow } from 'enzyme';
import Color from '../client/src/components/Color';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    url: data[0].colors[0].images[0],
  };
  const wrapper = shallow(<Color url={props.url}/>);
  return { wrapper, props };
}

describe('Color Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should set correct background from url', () => {
    expect(wrapper).toMatchSnapshot();
  });
});