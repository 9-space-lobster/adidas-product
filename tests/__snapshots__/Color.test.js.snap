// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component Test Suite It should set correct background from url 1`] = `
<styled.div>
  <styled.div
    url="B42200_color0_pic0.jpg"
  />
</styled.div>
`;
